### PR TITLE
PN-186: When a user contacts another user regarding a match using the Similar Patients UI, mark that match as 'notified' in the match table in the administration; PN-105: Contact form should only be available for the owner or manager of patient record; RM-55: Contact form should only be available for the owner or manager of patient record

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

File: ui/src/main/resources/PhenoTips/SimilarCases.xml [hunk 1]
```diff
@@ -485,12 +489,14 @@
 
     _displayResultContactInfo : function (r) {
       var contactType = (r.myCase || r.myGroupsCase) ? "none" : (r.access == "public" ? "open" : "restricted");
-      var icon = (contactType == 'restricted') ? this.Utils.generateIcon('contact', "Contact owner for further information") : this.Utils.generateIcon('group');
+      var icon = (contactType == 'restricted' &amp;&amp; this._data.isManager) ? this.Utils.generateIcon('contact', "Contact owner for further information") : this.Utils.generateIcon('group');
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: This is wrong, now for somebody that's not a manager of the patient, "this belongs to one of your groups" will show up, regardless whether the patient is actually accessible or not.

File: ui/src/main/resources/PhenoTips/SimilarCases.xml [hunk 2]
```diff
@@ -42,21 +42,25 @@
...
+#set ($isManager = $services.permissions.getPatientAccess($patient).hasAccessLevel('manage'))
 {
   "query": $patient.toJSON(),
+  "isManager" : $isManager,
   "resultsCount" : $matches.size(),
   "results" : [
 #foreach ($p in $matches)
-  $p.toJSON()#if ($foreach.hasNext()),
+  #set ($match = $p.toJSON())
+  #set ($match.owner.email = '')$match#if ($foreach.hasNext()),
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: Why?

File: communication/src/main/java/org/phenotips/messaging/script/AnonymousCommunicationScriptService.java [hunk 3]
```diff
@@ -102,4 +115,27 @@ public Connection grantAccess(String connectionId)
...
     }
+
+    /**
+     * When a user contacts another user regarding a match using the Similar Patients UI,
+     * mark that match as 'notified' in the match table in the administration.
+     *
+     * @param patientId reference patientID
+     * @param matchId matched patient ID
+     */
+    private void setNotified(String patientId, String matchId)
```
Comment (sdumitriu) [inline communication/src/main/java/org/phenotips/messaging/script/AnonymousCommunicationScriptService.java]: Why can't this be done inside `actionManager`? Script services shouldn't contain this much logic.

File: ui/src/main/resources/PhenoTips/SimilarCases.xml (hunk 2 above)
Comment (veronikaslc) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: stupidly forgot the main part of it `#if (!$isManager)`

File: ui/src/main/resources/PhenoTips/SimilarCases.xml (hunk 1 above)
Comment (veronikaslc) [inline ui/src/main/resources/PhenoTips/SimilarCases.xml]: not sure its clear for me..

**Pull request merged**